card/src/main/java/com/adyen/checkout/card/CardComponent.kt

```diff
@@ -189,23 +214,87 @@ class CardComponent private constructor(
             cardDelegate.validateSocialSecurityNumber(socialSecurityNumber),
             cardDelegate.validateKcpBirthDateOrTaxNumber(kcpBirthDateOrTaxNumber),
             cardDelegate.validateKcpCardPassword(kcpCardPassword),
-            cardDelegate.validatePostalCode(postalCode),
+            cardDelegate.validateAddress(addressInputModel, addressFormUIState),
             makeInstallmentFieldState(selectedInstallmentOption),
             isStorePaymentSelected,
             makeCvcUIState(selectedOrFirstCardType?.cvcPolicy),
             makeExpiryDateUIState(selectedOrFirstCardType?.expiryDatePolicy),
             outputCardTypes,
             cardDelegate.isSocialSecurityNumberRequired(),
             cardDelegate.isKCPAuthRequired(),
-            cardDelegate.isPostalCodeRequired(),
+            addressFormUIState,
             cardDelegate.getInstallmentOptions(
                 configuration.installmentConfiguration,
                 selectedOrFirstCardType?.cardType,
                 isReliable
-            )
+            ),
+            countryOptions,
+            stateOptions
         )
     }
 
+    private fun subscribeToStatesList() {
+        (cardDelegate as NewCardDelegate).stateListFlow
+            .distinctUntilChanged()
+            .onEach {
+                Logger.d(TAG, "New states emitted")
+                Logger.d(TAG, "States: $it")
+                with(outputData) {
+                    this ?: return@with
+                    val newOutputData = makeOutputData(
+                        cardNumber = cardNumberState.value,
+                        expiryDate = expiryDateState.value,
+                        securityCode = securityCodeState.value,
+                        holderName = holderNameState.value,
+                        socialSecurityNumber = socialSecurityNumberState.value,
+                        kcpBirthDateOrTaxNumber = kcpBirthDateOrTaxNumberState.value,
+                        kcpCardPassword = kcpCardPasswordState.value,
+                        addressInputModel = inputData.address,
+                        isStorePaymentSelected = isStoredPaymentMethodEnable,
+                        detectedCardTypes = detectedCardTypes,
+                        selectedCardIndex = inputData.selectedCardIndex,
+                        selectedInstallmentOption = null,
+                        countryOptions = countryOptions,
+                        stateOptions = AddressFormUtils.mapToListItem(it, true)
+                    )
+                    notifyStateChanged(newOutputData)
+                }
+            }
+            .launchIn(viewModelScope)
+    }
+
+    private fun requestCountryList() {
+        val newCardDelegate = cardDelegate as NewCardDelegate
```

Comment:
The IDE shows the `newCardDelegate` greyed out like it's unused. Maybe we could pass it as a param to `requestCountryList()` and also `subscribeToStatesList()` since we do already check `if (cardDelegate is NewCardDelegate)` in the block where we call them?